platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/isolated/FetchGradleProjectForTarget.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.configurationcache.isolated;
+
+import org.gradle.tooling.BuildAction;
+import org.gradle.tooling.BuildController;
+import org.gradle.tooling.model.GradleProject;
+import org.gradle.tooling.model.gradle.BasicGradleProject;
+import org.gradle.tooling.model.gradle.GradleBuild;
+
+public class FetchGradleProjectForTarget implements BuildAction<GradleProject> {
+
+    private final String targetProject;
+
+    public FetchGradleProjectForTarget(String targetProject) {
+        this.targetProject = targetProject;
+    }
+
+    @Override
+    public GradleProject execute(BuildController controller) {
+        GradleBuild buildModel = controller.getBuildModel();
+        for (BasicGradleProject project : buildModel.getProjects()) {
```

Comment:
💭 This loop isn't strictly necessarily, as the root build is included in the set of editable builds.

Reply:
Initially, I thought so as well, but this is not the case. This is also documented in the javadoc:
https://github.com/gradle/gradle/blob/866060514f01806785b1aef1c029dc402175f41a/platforms/ide/tooling-api/src/main/java/org/gradle/tooling/model/gradle/GradleBuild.java#L71-L75

Reply:
huh, I added that comment.

Reply:
To be clear, it's not just the comment. It actually works that way. If I remove this loop, then the tests requesting projects from the root build start failing.